x-pack/test/security_solution_endpoint_api_int/apis/resolver/entity_id.ts

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import expect from '@kbn/expect';
+import { SearchResponse } from 'elasticsearch';
+import { eventsIndexPattern } from '../../../../plugins/security_solution/common/endpoint/constants';
+import {
+  ResolverTree,
+  ResolverEntityIndex,
+} from '../../../../plugins/security_solution/common/endpoint/types';
+import { FtrProviderContext } from '../../ftr_provider_context';
+import {
+  EndpointDocGenerator,
+  Event,
+} from '../../../../plugins/security_solution/common/endpoint/generate_data';
+import { InsertedEvents } from '../../services/resolver';
+
+export default function resolverAPIIntegrationTests({ getService }: FtrProviderContext) {
+  const supertest = getService('supertest');
+  const resolver = getService('resolverGenerator');
+  const es = getService('es');
+  const generator = new EndpointDocGenerator('resolver');
+
+  describe('Resolver handling of entity ids', () => {
+    describe('entity api', () => {
+      let origin: Event;
+      let genData: InsertedEvents;
+      before(async () => {
+        origin = generator.generateEvent({ parentEntityID: 'a' });
+        origin.process.entity_id = '';
+        genData = await resolver.insertEvents([origin]);
+      });
+
+      after(async () => {
+        await resolver.deleteData(genData);
+      });
+
+      it('excludes events that have an empty entity_id field', async () => {
+        // first lets get the _id of the document using the parent.process.entity_id
```

Comment:
nothing to block over, but can't you set `_id` when you insert a document?

Reply:
Yeah good point. I'm using a helper function that inserts an array of documents so it's doing a bulk request. Later we can refactor it so it specifies the `_id` of the document so we don't have to do the search.